src/layers.c

```diff
@@ -29,44 +45,9 @@ layers_arrange(struct output *output)
 	struct wlr_box usable_area = full_area;
 	struct wlr_box old_usable_area = output->usable_area;
 
-	struct server *server = output->server;
-	struct wlr_scene_output *scene_output =
-		wlr_scene_get_scene_output(server->scene, output->wlr_output);
-	if (!scene_output) {
-		wlr_log(WLR_DEBUG, "no wlr_scene_output");
-		return;
-	}
-
 	int nr_layers = sizeof(output->layers) / sizeof(output->layers[0]);
 	for (int i = 0; i < nr_layers; i++) {
-		struct lab_layer_surface *lab_layer_surface;
-
-		/*
-		 * First we go over the list of surfaces that have
-		 * exclusive_zone set (e.g. statusbars) because we have to
-		 * determine the usable area before processing regular layouts.
-		 */
-		wl_list_for_each(lab_layer_surface, &output->layers[i], link) {
-			struct wlr_scene_layer_surface_v1 *scene_layer_surface =
-				lab_layer_surface->scene_layer_surface;
-			if (scene_layer_surface->layer_surface->current.exclusive_zone) {
-				wlr_scene_layer_surface_v1_configure(
-					scene_layer_surface, &full_area, &usable_area);
-			}
-		}
-
-		/* Now we process regular layouts */
-		wl_list_for_each(lab_layer_surface, &output->layers[i], link) {
-			struct wlr_scene_layer_surface_v1 *scene_layer_surface =
-				lab_layer_surface->scene_layer_surface;
-			if (!scene_layer_surface->layer_surface->current.exclusive_zone) {
-				wlr_scene_layer_surface_v1_configure(
-					scene_layer_surface, &full_area, &usable_area);
-			}
-		}
-
-		wlr_scene_node_set_position(&output->layer_tree[i]->node,
-			scene_output->x, scene_output->y);
```

Comment:
I think we still need to adjust the whole layer tree here to react to output resolution changes

Reply:
Oh. Good spot. Shouldn't have taken that line out.

Reply:
I've put `wlr_scene_node_set_position()` back :smile:

Reply:
Might deserve a comment about resolution changes while at it :)